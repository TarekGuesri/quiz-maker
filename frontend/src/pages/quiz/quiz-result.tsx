import React from "react";
import Typography from "@mui/material/Typography";
import Box from "@mui/material/Box";
import { CircularProgressWithLabel } from "src/components/progress/circular-progress-with-label";

interface QuizResultProps {
  score: number;
}

export const QuizResult: React.FC<QuizResultProps> = ({ score }) => {
  return (
    <>
      <Typography color="secondary" variant="h4" component="div" mb={4}>
        Quiz Result
      </Typography>

      <Box mb={4}>
        <CircularProgressWithLabel value={score} />
      </Box>

      <Typography variant="h6" mb={1}>
        You have completed the quiz.
      </Typography>
      <Typography variant="h6">
        You answered <strong> {score}%</strong> of the questions correctly.
      </Typography>
    </>
  );
};
