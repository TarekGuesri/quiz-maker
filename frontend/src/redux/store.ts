import { Action, ThunkAction, configureStore } from "@reduxjs/toolkit";

import ui from "./ui/ui-slice";

export const store = configureStore({
  reducer: {
    ui,
  },
  devTools: true,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
