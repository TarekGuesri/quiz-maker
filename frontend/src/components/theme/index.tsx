import React, { FC } from 'react';
import CssBaseline from '@mui/material/CssBaseline';
import PropTypes from 'prop-types';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { useAppSelector } from 'src/redux/hooks';
import varelaRound from 'src/assets/fonts/VarelaRound-Regular.ttf';

export const Theme: FC = (props) => {
  const { darkMode } = useAppSelector((state) => state.ui);

  const theme = createTheme({
    palette: {
      mode: darkMode ? 'dark' : 'light',
      background: {
        default: darkMode ? '#333' : '#f7f7f7',
        paper: darkMode ? '#333' : '#fbfbfb',
      },
      primary: {
        main: '#8e49de',
        // contrastText:
      },
      secondary: {
        main: '#f9de4a',
      },
      contrastThreshold: 3,
      tonalOffset: 0.2,
    },
    typography: {
      fontFamily: 'Varela Round',
    },
    components: {
      MuiCssBaseline: {
        styleOverrides: `
          @font-face {
            font-family: 'Varela Round';
            font-style: normal;
            font-display: swap;
            font-weight: 400;
            src: url(${varelaRound}) format('truetype');
        `,
      },
    },
  });

  return (
    <ThemeProvider theme={theme}>
      {props.children}
      <CssBaseline />
    </ThemeProvider>
  );
};

Theme.propTypes = {
  children: PropTypes.node.isRequired,
};
