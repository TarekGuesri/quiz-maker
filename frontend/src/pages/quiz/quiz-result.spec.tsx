import React from "react";
import { cleanup, render } from "@testing-library/react";
import { store } from "src/redux/store";
import { Provider } from "react-redux";
import { Theme } from "src/components/theme";
import { QuizResult } from "./quiz-result";

describe("src/pages/quiz/quiz-result.tsx", () => {
  afterEach(() => cleanup());

  test("Render QuizResult component", () => {
    const renderedComponent = render(
      <Provider store={store}>
        <Theme>
          <QuizResult score={50} />
        </Theme>
      </Provider>,
    );

    expect(renderedComponent.container).toMatchSnapshot();
  });
});
