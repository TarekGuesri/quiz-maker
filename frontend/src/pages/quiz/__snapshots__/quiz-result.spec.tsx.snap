// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/pages/quiz/quiz-result.tsx Render QuizResult component 1`] = `
<div>
  <div
    class="MuiTypography-root MuiTypography-h4 css-mzhb8u-MuiTypography-root"
  >
    Quiz Result
  </div>
  <div
    class="MuiBox-root css-1yjvs5a"
  >
    <div
      class="MuiBox-root css-rmz4vk"
    >
      <span
        aria-valuenow="100"
        class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorSecondary css-5nap9f-MuiCircularProgress-root"
        role="progressbar"
        style="width: 160px; height: 160px; transform: rotate(-90deg);"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
            style="stroke-dasharray: 126.920; stroke-dashoffset: 0.000px;"
          />
        </svg>
      </span>
      <span
        aria-valuenow="50"
        class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1pzb5l9-MuiCircularProgress-root"
        role="progressbar"
        style="width: 160px; height: 160px; transform: rotate(-90deg);"
      >
        <svg
          class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
            cx="44"
            cy="44"
            fill="none"
            r="20.2"
            stroke-width="3.6"
            style="stroke-dasharray: 126.920; stroke-dashoffset: 63.460px;"
          />
        </svg>
      </span>
      <div
        class="MuiBox-root css-1o8dslu"
      >
        <div
          class="MuiTypography-root MuiTypography-h4 css-1pxewg2-MuiTypography-root"
        >
          50%
        </div>
      </div>
    </div>
  </div>
  <h6
    class="MuiTypography-root MuiTypography-h6 css-1y2cxo8-MuiTypography-root"
  >
    You have completed the quiz.
  </h6>
  <h6
    class="MuiTypography-root MuiTypography-h6 css-xq7e3s-MuiTypography-root"
  >
    You answered 
    <strong>
       
      50
      %
    </strong>
     of the questions correctly.
  </h6>
</div>
`;
